src/main/java/org/kuali/kfs/module/purap/document/web/struts/PurchasingActionBase.java

```diff
@@ -1647,9 +1655,11 @@ private void createReasonNote(ActionForm form) {
     	KualiDocumentFormBase kualiDocumentFormBase = (KualiDocumentFormBase) form;
         try {
             Note noteObj = SpringContext.getBean(DocumentService.class).createNoteFromDocument(kualiDocumentFormBase.getDocument(), ((PurchasingFormBase)kualiDocumentFormBase).getReasonToChange());
+            populateIdentifierOnNoteAndExtension(noteObj);
             kualiDocumentFormBase.getDocument().addNote(noteObj);
-         //   SpringContext.getBean(NoteService.class).save(noteObj);
-            SpringContext.getBean(NoteService.class).saveNoteList(kualiDocumentFormBase.getDocument().getNotes());
+            if (isNoteReadyToBeSaved(kualiDocumentFormBase.getDocument(), noteObj)) {

```

Comment:
Thank you Chad.

It seems that the isNoteReadyToBeSaved is checking the document to see if it is in a state where a note for it can be saved and not necessarily if the note itself is in a state where it can be saved. Maybe the method name could be improved.

I also noticed that the new code is only saving the reason not and the previous code saved all the notes:
 SpringContext.getBean(NoteService.class).saveNoteList(kualiDocumentFormBase.getDocument().getNotes());

Was the saving of all notes unnecessary?

This particular customization for the send to vendor flag could be improved for clarity and for easier maintenance in the future. However that would be a significant effort and is out of the scope of this particular fix.

I think that if you can find a better name for the isNoteReadyToBeSaved method and ensure saving all notes don't need to be saved then we can promote this fix to DEV for validation.